src/Native/Unix/System.Native/pal_time.c

```diff
@@ -48,75 +48,43 @@ int32_t SystemNative_UTimensat(const char* path, TimeSpec* times)
     return result;
 }
 
-int32_t SystemNative_GetTimestampResolution(uint64_t* resolution)
+uint64_t SystemNative_GetTimestampResolution()
 {
-    assert(resolution);
-
 #if HAVE_MACH_ABSOLUTE_TIME
     mach_timebase_info_data_t mtid;
-    if (mach_timebase_info(&mtid) == KERN_SUCCESS)
-    {
-        *resolution = SecondsToNanoSeconds * ((uint64_t)(mtid.denom) / (uint64_t)(mtid.numer));
-        return 1;
-    }
-    else
-    {
-        *resolution = 0;
-        return 0;
-    }
 
-#elif HAVE_CLOCK_MONOTONIC
+    kern_return_t result = mach_timebase_info(&mtid);
+    assert(result == KERN_SUCCESS);
+    (void)result; // suppress unused parameter warning in release builds
+
+    return SecondsToNanoSeconds * ((uint64_t)(mtid.denom) / (uint64_t)(mtid.numer));
+#else
     // Make sure we can call clock_gettime with MONOTONIC.  Stopwatch invokes
     // GetTimestampResolution as the very first thing, and by calling this here
     // to verify we can successfully, we don't have to branch in GetTimestamp.
-    struct timespec ts;
-    if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) 
```

Comment:
Given the above comment (which you've left in), why is it ok to remove this branching?

Reply:
Fixed, and updated to appropriately call `clock_getres` since the value is not guaranteed.